actions/views/drafts.ts

```diff
@@ -28,15 +29,36 @@ type Draft = {
 }
 
 export function getDrafts(teamId: string) {
-    return async (dispatch: DispatchFunc) => {
-        let drafts: ServerDraft[] = [];
-        drafts = await Client4.getUserDrafts(teamId);
-        const actions = (drafts || []).map((draft) => {
-            const {key, value} = transformServerDraft(draft);
-            return setGlobalItem(key, value);
+    const getLocalDrafts = makeGetDrafts();
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState() as GlobalState;
+
+        let serverDrafts: Draft[] = [];
+        if (syncedDraftsAreAllowedAndEnabled(state)) {
+            serverDrafts = (await Client4.getUserDrafts(teamId) || []).map((draft) => transformServerDraft(draft));
+        }
+
+        const localDrafts = getLocalDrafts(state);
+        const drafts = [...serverDrafts, ...localDrafts];
+
+        if (drafts === null) {
+            return {data: false};
+        }
+
+        const draftsMap = new Map((drafts || []).map((draft) => [draft.key, draft]));
+        drafts.forEach((draft) => {
+            const currentDraft = draftsMap.get(draft.key);
+            if (currentDraft && draft.timestamp > currentDraft.timestamp) {
+                draftsMap.set(draft.key, draft);
+            }
+        });
+
+        const actions = Array.from(draftsMap || []).map(([key, draft]) => {
```

Comment:
Or here. We're also mixing types in this one because `draftsMap` is a Map and not an array